### PR TITLE
Scalar-terraform and scalar-k8s resource usage matrix

### PR DESCRIPTION
**Pull request by scalar-boney**
Scalar-terraform and scalar-k8s resource usage matrix created.

### CODE REVIEW

**Review by feeblefakie — COMMENTED** [review]
Overall looking good. Left some comments.

File: docs/ResourceUsage.md [hunk 1]
```diff
@@ -0,0 +1,370 @@
...
+
+| scalar-terraform (chiku) | Network | Cassandra | Scalardl | Envoy | Monitor | Cassy | Reaper | Total |
+|:-------------------------|:--------|:----------|:---------|:------|:--------|:------|:-------|:------|
+| OS Disk (Premium SSD) | 64 GB |  192 GB (64 GB per disk) |  192 GB (64 GB per disk) |  192 GB (64 GB per disk) | 64 GB | 64 GB | 64 GB | 832 GB |
+| Data Disk (Premium SSD / Standard HDD) | 0 | 3072 GB (1024 GB per disk) (Premium SSD) | 0 | 0 | 500 GB (Standard HDD) | 0 | 0 | 3572 GB|
+| Total | 64 GB | 3264 GB | 192 GB | 192 GB | 564 GB | 64 GB | 64 GB | 4404 GB |
+
+**Resource Price for Japan East Region** 
+
+Base: _chiku_
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ResourceUsage.md]: Thank you for your review comments.
> Is it needed?

No,
The document has been updated accordingly.

File: docs/ResourceUsage.md [hunk 2]
```diff
@@ -0,0 +1,370 @@
...
+| Envoy LB | 1 | $0 | $0 | $0 | $18.26 (Assuming 1Gb data processed) | $18.26 |
+| Total | 14 | $1152.38 | $152.62 | $488.08 | $18.26 | $1811.34 |
+
+Note: 
+* _Load balancer cost will change according to network traffic._
+* _Blob storage cost for cassy backup :- $0.02 per GB (For first 50 TB)_
+ 
+**Resource Price for West US 2 Region** 
+
+Base: _chiku_
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ResourceUsage.md]: The document has been updated accordingly.

File: docs/ResourceUsage.md [hunk 3]
```diff
@@ -0,0 +1,370 @@
...
+
+| scalar-terraform (bai) | Network | Cassandra | Scalardl | Envoy | Monitor | Cassy | Reaper | Total |
+|:-----------------------|:--------|:----------|:---------|:------|:--------|:------|:-------|:------|
+| OS Disk (Premium SSD) | 64 GB |  192 GB (64 GB per disk) |  192 GB (64 GB per disk) |  192 GB (64 GB per disk) | 64 GB | 64 GB | 64 GB | 832 GB |
+| Data Disk (Premium SSD / Standard HDD) | 0 | 3072 GB (1024 GB per disk) (Premium SSD) | 0 | 0 | 500 GB (Standard HDD) | 0 | 0 | 3572 GB|
+| Total | 64 GB | 3264 GB | 192 GB | 192 GB | 564 GB | 64 GB | 64 GB | 4404 GB |
+
+**Resource Price for Japan East Region**
+
+Base: _bai_
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ResourceUsage.md]: The document has been updated accordingly.

File: docs/ResourceUsage.md [hunk 4]
```diff
@@ -0,0 +1,370 @@
...
+| Envoy LB | 1 | $0 | $0 | $0 | $18.26 (Assuming 1Gb data processed) | $18.26 |
+| Total | 14 | $801.98 | $152.62 | $488.08 | $18.26 | $1460.94 |
+
+Note: 
+* _Load balancer cost will change according to network traffic._
+* _Blob storage cost for cassy backup :- $0.02 per GB (For first 50 TB)_
+
+**Resource Price for West US 2 Region**
+
+Base: _bai_
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ResourceUsage.md]: The document has been updated accordingly.

File: docs/ResourceUsage.md [hunk 5]
```diff
@@ -0,0 +1,370 @@
+# Resource Usage
+
+Scalar-terraform and scalar-k8s resource usage and price details available here. Resource usage price calculated on monthly basis.
```
[the remark this answers is not in the document]
Comment (scalar-boney) [reply docs/ResourceUsage.md]: Thank you for the review comment.
The document has been updated accordingly.

File: docs/ResourceUsage.md [hunk 6]
```diff
@@ -0,0 +1,346 @@
+# Resource Usage
+
+This document shows the resource usage and the monthly prices for the resources used by scalar-terraform and scalar-k8s.
+
+## Scalar-terraform resource usage
```
Comment (feeblefakie) [inline docs/ResourceUsage.md:5]: It should note that the numbers are from the default settings.

File: docs/ResourceUsage.md [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+
+### Azure
+
+#### **Chiku**
```
Comment (feeblefakie) [inline docs/ResourceUsage.md:9]: Indenting is too deep. 
Why don't you separate the doc into 2 files; terraform and k8s.

**Review by feeblefakie — CHANGES_REQUESTED** [review]

File: docs/ResourceUsage.md (hunk 6 above)
Comment (scalar-boney) [reply docs/ResourceUsage.md:5]: @feeblefakie san,
Sorry,
In the AWS scalar-terraform, the default `bastion_resource_count` is 2 but here we calculated the cost for only one bastion resource.

Is it need to calculate the cost for 2 bastion resources?

In Azure scalar-terraform, the default value of  `bastion_resource_count` is 1

File: docs/ResourceUsage.md (hunk 7 above)
Comment (scalar-boney) [reply docs/ResourceUsage.md]: Thank you for your suggestion.
The resource usage matrix is ​​split into 2.
- [TerraformResourceUsage.md](https://github.com/scalar-labs/scalar-terraform/blob/resource_usage_matrix/docs/TerraformResourceUsage.md)
- [K8SResourceUsage.md](https://github.com/scalar-labs/scalar-terraform/blob/resource_usage_matrix/docs/K8SResourceUsage.md)

**Review by feeblefakie — APPROVED** [review]
LGTM. Thanks.

**Pull request merged**